client/packages/system/src/Stock/Components/Campaign/CampaignOrProgramSelector.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ReactElement, SyntheticEvent } from 'react';
+import { useCampaigns } from '@openmsupply-client/system/src/Manage/Campaigns/api';
+import { Autocomplete, AutocompleteOption } from '@common/components';
+import { CampaignNode } from '@common/types';
+import { useTranslation } from '@common/intl';
+import { ProgramFragment, useProgramList } from '@openmsupply-client/programs';
+
+enum OptionType {
+  Campaign = 'campaign',
+  Program = 'program',
+}
+
+interface CampaignOrProgramOption {
+  label: string;
+  value: string | null;
+  type: OptionType;
+}
+
+interface CampaignOrProgramSelectorProps {
+  campaignId?: string;
+  programId?: string;
+  itemId: string;
+  onChange: (value: {
+    campaign: CampaignNode | null;
+    program: ProgramFragment | null;
+  }) => void;
+}
+
+export const CampaignOrProgramSelector = ({
```

Comment:
Ah this is actually just the edited CampaignSelector - with added handling for if program or campaign